### PR TITLE
RHELPLAN-41572 - Logging - Convert each subrole to a pair of task/var files

### PR DESCRIPTION
**Pull request by nhosoi**
Apply the idea of "Convert the files output subrole to a pair of task/var files" by @pcahyna to the rest of the subroles.

```
commit 4b77713745569589ef966c13825495c7846a9cae
Author: Pavel Cahyna <pcahyna@redhat.com>
Date:   Fri Apr 24 23:49:38 2020 +0200

    Convert the files output subrole to a pair of task/var files
    
    UNTESTED
    
    Remove some debug stuff in the task file
```

### CODE REVIEW

**Comment by richm** [comment]
lgtm

**Comment by nhosoi** [comment]
Thanks for reviewing the pr, @richm.  I'm updating the doc now...  When submitting a PR in linux-system-roles/logging (once pr100 is merged), I'm appending the doc commit, too.  Thanks!

File: roles/rsyslog/tasks/main.yml [hunk 1]
```diff
@@ -130,37 +130,80 @@
...
-    - name: Run remove input sub-roles configs
-      include_role:
-        name: "{{ role_path }}/roles/input_roles/{{ input_item.type }}"
-        tasks_from: cleanup.yml
+    - name: Run remove input sub-tasks
+      include_tasks:
+        file: "{{ tasks }}"
+      vars:
+        tasks: "inputs/{{ input_item.type }}_cleanup.yml"
+        tasks_abspath: "{{ role_path }}/tasks/{{ tasks }}"
```
Comment (pcahyna) [inline roles/rsyslog/tasks/main.yml]: what is this variable used for?

File: roles/rsyslog/tasks/main.yml [hunk 2]
```diff
@@ -130,37 +130,80 @@
...
 
-    - name: Run remove input sub-roles configs
-      include_role:
-        name: "{{ role_path }}/roles/input_roles/{{ input_item.type }}"
-        tasks_from: cleanup.yml
+    - name: Run remove input sub-tasks
+      include_tasks:
+        file: "{{ tasks }}"
+      vars:
+        tasks: "inputs/{{ input_item.type }}_cleanup.yml"
```
Comment (pcahyna) [inline roles/rsyslog/tasks/main.yml]: I would create a subdirectory for every former input subrole, so make `inputs/{{ input_item.type }}_cleanup.yml` into `inputs/{{ input_item.type }}/cleanup.yml` (and `inputs/{{ input_item.type }}.yml` into `inputs/{{ input_item.type }}/main.yml`)

File: roles/rsyslog/tasks/main.yml [hunk 3]
```diff
@@ -130,37 +130,80 @@
...
+      vars:
+        varfile: "outputs/{{ output_item.type }}.yml"
+        varfile_abspath: "{{ role_path }}/vars/{{ varfile }}"
+      loop: "{{ rsyslog_outputs | flatten(1) }}"
+      loop_control:
+        loop_var: output_item
+      when:
+        - rsyslog_enabled | bool
+        - output_item | d([])
+        - varfile_abspath is exists
```
Comment (pcahyna) [inline roles/rsyslog/tasks/main.yml]: is it needed? Are there outputs which do not have a variable file?

File: roles/rsyslog/tasks/main.yml [hunk 4]
```diff
@@ -130,37 +130,80 @@
...
+      vars:
+        tasks: "outputs/{{ output_item.type }}.yml"
+        tasks_abspath: "{{ role_path }}/tasks/{{ tasks }}"
       loop: "{{ rsyslog_outputs | flatten(1) }}"
       loop_control:
         loop_var: output_item
       when:
         - rsyslog_enabled | bool
         - output_item | d([])
+        - tasks_abspath is exists
```
Comment (pcahyna) [inline roles/rsyslog/tasks/main.yml]: is it needed? Are there outputs which do not have a task file?

File: roles/rsyslog/tasks/main.yml [hunk 5]
```diff
@@ -130,37 +130,80 @@
...
+        - varfile_abspath is exists
+      tags:
+        - skip_ansible_lint
+
+    - name: Run input sub-tasks 
+      include_tasks:
+        file: "{{ tasks }}"
+      vars:
+        tasks: "inputs/{{ input_item.type }}.yml"
+        tasks_abspath: "{{ role_path }}/tasks/{{ tasks }}"
```
Comment (pcahyna) [inline roles/rsyslog/tasks/main.yml]: is it needed? Are there inputs which do not have a task file?

File: roles/rsyslog/tasks/main.yml [hunk 6]
```diff
@@ -130,37 +130,80 @@
...
+      loop: "{{ logging_inputs | flatten(1) }}"
+      loop_control:
+        loop_var: input_item
+      when:
+        - rsyslog_enabled | bool
+        - input_item | d([])
+        - tasks_abspath is exists
```
Comment (pcahyna) [inline roles/rsyslog/tasks/main.yml]: is it needed? Are there inputs which do not have a task file?

File: roles/rsyslog/tasks/main.yml [hunk 7]
```diff
@@ -130,37 +130,80 @@
...
+        varfile: "inputs/{{ input_item.type }}.yml"
+        varfile_abspath: "{{ role_path }}/vars/{{ varfile }}"
       loop: "{{ logging_inputs | flatten(1) }}"
       loop_control:
         loop_var: input_item
       when:
         - rsyslog_enabled | bool
         - input_item | d([])
-        - input_item.state | d('present') != 'absent'
+        - varfile_abspath is exists
```
Comment (pcahyna) [inline roles/rsyslog/tasks/main.yml]: is it needed? Are there inputs which do not have a vars file?

File: roles/rsyslog/tasks/main.yml (hunk 1 above)
Comment (nhosoi) [reply roles/rsyslog/tasks/main.yml]: Thanks!  I missed to add this condition...
`- tasks_abspath is exists`

File: roles/rsyslog/tasks/main.yml (hunk 3 above)
Comment (nhosoi) [reply roles/rsyslog/tasks/main.yml]: A good question.  Probably, not. :)  Let me remove the line.

File: roles/rsyslog/tasks/main.yml (hunk 4 above)
Comment (nhosoi) [reply roles/rsyslog/tasks/main.yml]: Ok...  So, you are suggesting to get rid of {tasks,varfile}_abspath completely...

Comment (pcahyna) [reply roles/rsyslog/tasks/main.yml]: yes, if there ever is an input or output which lacks a var file or task file, it is probably simpler to add a dummy empty one.

**Pull request closed without merging**